### DISCUSSION TITLE
Configuration: ENV_VARs, .yaml/.toml, or some secret third thing?

### DISCUSSION THREAD

**Discussion by bobmcwhirter** [guacsec/trustify#34] [Ideas]
Configuration is hard, particularly to accommodate different use-cases.

It gets harder when/if we provide any `admin` type UI to manipulate that configuration.

I suspect we'll need a mixture of "configuration" types:

1. Those related to running The System (wtf is my DB?)
2. Those related to options within a running System (which walkers are enabled, etc)

**Comment by ctron** [comment]
I think that every that potentially gets configuring during runtime should end up in the database, walker jobs included. For some it might be ok to have a generic "configuration" table (or not!) but for some (like the walker jobs) it actually makes sense to have more specializes entities, which also carry some state (like last time run, import report, total time spent, …)

**Reply by bobmcwhirter** [reply]
agreed 100%
